youtube_dl/extractor/lcp.py

```diff
@@ -0,0 +1,122 @@
+        media_files_info = self.__extract_from_webpage(display_id, webpage)
+        # Some web pages embed videos from other platforms like dailymotion, therefore we pass on these URLs
+        if media_files_info is None:
+            return self.url_result(url, 'Generic')
+
+        video_formats = self.__get_video_formats(media_files_info)
+        video_thumbnails = self.__get_thumbnails(media_files_info)
+
+        return {
+            'id': media_files_info['EntryName'],
+            'title': self._og_search_title(webpage),
+            'formats': video_formats,
+            'thumbnails': video_thumbnails
+        }
+
+    def __extract_from_webpage(self, display_id, webpage):
+        """Extracts the media info JSON object for the video for the provided web page."""
+        embed_url = self.__extract_embed_url(webpage)
+        embed_regex = r'(?:[a-zA-Z0-9]+\.)?lcp\.fr/embed/(?P<clip_id>[A-za-z0-9]+)/(?P<player_id>[A-za-z0-9]+)/(?P<skin_name>[^\/]+)'
+
+        clip_id = self._search_regex(embed_regex, embed_url, 'clip id', group='clip_id', fatal=False)
+        player_id = self._search_regex(embed_regex, embed_url, 'player id', group='player_id', fatal=False)
+        skin_name = self._search_regex(embed_regex, embed_url, 'skin name', group='skin_name', fatal=False)
+
+        # Check whether the matches failed, which might be when dealing with other players (e.g., dailymotion stream)
+        if (clip_id is None) or (player_id is None) or (skin_name is None):
+            return None
+
+        return self.__extract_from_player(display_id, clip_id, player_id, skin_name)
+
+    def __extract_embed_url(self, webpage):
+        return self._search_regex(
+            r'<iframe[^>]+src=(["\'])(?P<url>.+?)\1',
+            webpage, 'embed url', group='url')
+
+    def __extract_from_player(self, display_id, clip_id, player_id, skin_name):
+        """Extracts the JSON object containing the required media info from the embedded arkena player"""
+        arkena_url = 'http://play.arkena.com/config/avp/v1/player/media/{0}/{1}/{2}/?callbackMethod=?'.format(clip_id,
+                                                                                                              skin_name,
+                                                                                                              player_id)
+        arkena_info = self._download_webpage(arkena_url, 'clip_info_' + clip_id)
+
+        arkena_info_regex = r'\?\((?P<json>.*)\);'
+        info_json = self._parse_json(self._search_regex(arkena_info_regex, arkena_info, 'json', group='json'),
+                                     display_id)
+
+        # All videos are part of a playlist, a single video is also put in a playlist
+        media_files_info = info_json.get('Playlist')
+        if media_files_info is not None:
+            media_files_info = media_files_info[0]
+        return media_files_info
+
+    def __get_thumbnails(self, media_files_info):
```

Comment:
Extracting code in separate methods is questionable in this case since all of them are used only once and just make it harder to read the code due to jumping back and forth.